protocol/superchainERC20/convert.md

```diff
@@ -0,0 +1,176 @@
+- `convertFromSuper(address _*legacyAddr, address superAddr, uint256 _amount)`: converts* `_amount` *`SuperchainERC20` with address `_super*Addr` to the same `_amount` of legacy tokens in address `_legacyAddr`.
+
+These functions will:
+
+1. Check that the input and output addresses are valid.
+2. `burn` the input address balance from the caller.
+3. `mint` the output address balance to the caller.
+
+An example implementation that depends on the `isValidPair()` access control function looks like this:
+
+```solidity
+function convertToSuper(address _legacyAddr, address _superAddr, uint256 _amount) public {
+  require(isValidPair(_legacyAddr, _superAddr), "Invalid address pair");
+
+  IERC20(_legacyAddr).burn(msg.sender, _amount);
+  IERC20(_superAddr).mint(msg.sender, _amount);
+}
+
+function convertFromSuper(address _legacyAddr, address _superAddr, uint256 _amount) public {
+  require(isValidPair(_legacyAddr, _superAddr), "Invalid address pair");
+
+  IERC20(_superAddr).burn(msg.sender, _amount);
+  IERC20(_legacyAddr).mint(msg.sender, _amount);
+}
+```
+
+The corresponding `SuperchainERC20` will need to allow the `L2StandardBridge` to mint and burn tokens. We will discuss the `isValidPair()` function in the [section below](#access-control).
+
+> 💡 Note:
+> We considered placing the `convert()` function directly in the `SuperchainERC20`, similar to how WETH manages deposits and withdrawals. However, we believe including them in the `L2StandardBridge` makes more sense. This is because:
+>
+> - It makes the token leaner. Users will interact with `convert()` only a few times and in a context similar to deposit and withdraw.
+> - The `SuperchainERC20` representation for migrated liquidity will be as close as possible to newly deployed `SuperchainERC20`s.
+> - The `L2StandardBridge` needs to be involved because of burn and mint. Having the functionality in the token would have added additional external calls.
+
+### Access Control
+
+The contract will need to:
+
+- **Check `OptimismMintableERC20` address is valid**: there is no need to verify anything as any liquidity risk will be isolated.
+- **Check `SuperchainERC20` is valid**: There are two ways to implement access control in the Isolated design.
+  - **Registry**: If the Factory stores the `SuperchainERC20` address for each legacy address.
+    ```solidity
+    ISuperchainERC20Factory public superFactory = ISuperchainERC20Factory(factoryAddr);
+
+    function isValidPair(address _legacyAddr, address _superAddr) internal view returns (bool) {
+      return superFactory.deployment(_legacyAddr) == _superAddr;
+    }
+    ```
+  - **Deterministic address**: The `SuperchainERC20` factory will use the L2 representation address as a salt, so the address of the `SuperchainERC20` can be deduced from the address of the `OptimismMintableERC20` and the Factory (predeploy).
```

Comment:
Do we have any L2 chains with conflicting tokens at the same address, minted by the system factory? I remember Base running into something like this with a previous factory. If yes, then that breaks the deterministic-address assumption. Using the L1-address of the token may be more safe (if that can be authenticated to be correct).